run-checks

```diff
@@ -1,63 +1,94 @@
 #!/bin/sh
 
-set -e
+set -eu
 
 if which goctest >/dev/null; then
     goctest="goctest"
 else
     goctest="go test"
 fi
 
-echo Checking formatting
-fmt=$(gofmt -l .)
-
-if [ -n "$fmt" ]; then
-    echo "Formatting wrong in following files"
-    echo $fmt
-    exit 1
+STATIC=""
+UNIT_GO=""
+UNIT_NODE=""
+
+case "${1:-all}" in
+    all)
+        STATIC="yes"
+        UNIT_GO="yes"
+        UNIT_NODE="yes"
+        ;;
+    --static)
+        STATIC="yes"
+        ;;
+    --unit)
+        UNIT_GO="yes"
+        UNIT_NODE="yes"
+        ;;
+    --unit-go)
+        UNIT_GO="yes"
+        ;;
+    --unit-node)
+        UNIT_NODE="yes"
+        ;;
+    *)
+        echo "Wrong flat ${1}. To run a single suite use --static, --unit, --unit-go or --unit-node."
+        exit1
+esac
+
+if [ ! -z "$STATIC" ] || [ ! -z "$UNIT_GO" ]; then
+    sh ./get-go-deps.sh
```

Comment:
This was split in snapd because it was needed for autopkgtests. In here we don't need it yet, so I can return it to runchecks if you prefer that.